syllabus/curricula/plr/task_sampler.py

```diff
@@ -261,12 +316,40 @@ def _sample_unseen_level(self):
 
         return task
 
+    def _evaluate_unseen_level(self):
+        task_idx = \
+        np.random.choice(range(self.num_tasks), 1, p=self.unseen_task_weights / self.unseen_task_weights.sum())[0]
+        task = self.tasks[task_idx]
+        episode_data = self.evaluate_task(task, self.eval_envs, self.get_action_and_value_fn, self.gamma, self.gae_lambda)
+        self.update_with_episode_data(episode_data, self._average_gae)  # Update task scores
+        return task
+
+    def get_action_and_value_fn(agent_model, device):
+        def action_value_fn(obs):
+            # Convert observation to tensor if necessary
+            obs_tensor = torch.tensor(obs, dtype=torch.float32).unsqueeze(0).to(device)
+
+            # Forward pass through the agent's model to get action logits and state value
+            with torch.no_grad():
+                action_logits, state_value = agent_model(obs_tensor)
+
+            # Convert action logits to probabilities
+            action_probs = torch.softmax(action_logits, dim=-1)
+
+            # Sample action from the action probabilities
+            action = torch.multinomial(action_probs, num_samples=1).squeeze().item()
+
+            # Return the sampled action and the state value
+            return action, state_value.item()
+
+        return action_value_fn
```

Comment:
Lets assume the user will provide this function, make it an argument that gets passed to the curriculum